src/EditorFeatures/Core/Implementation/InlineRename/InlineRenameSession.cs

```diff
@@ -171,7 +172,9 @@ private void CreateOpenTextBufferManagerForBuffer(ITextBuffer buffer, IEnumerabl
             AssertIsForeground();
             VerifyNotDismissed();
 
-            if (!_openTextBuffers.ContainsKey(buffer))
+            var documentSupportsRefactoringService = _workspace.Services.GetService<IDocumentSupportsSuggestionService>();
+
+            if (!_openTextBuffers.ContainsKey(buffer) && documents.All(d => documentSupportsRefactoringService.SupportsRename(d)))
```

Comment:
Fixed by renaming `CreateOpenTextBufferManagerForBuffer` to `TryCreateOpenTextBufferManagerForBuffer` and only connect if we are actually tracking that text buffer.